samples/tmo_shell/src/tmo_shell.c

```diff
@@ -1760,22 +1782,41 @@ int cmd_dfu_get_version(const struct shell *shell, size_t argc, char **argv)
 
 int cmd_dfu_update(const struct shell *shell, size_t argc, char **argv)
 {
-	if (argc < 3){
+	int firmware_target = (int) strtol(argv[1], NULL, 10);
+	int delta_firmware_target = (int) strtol(argv[2], NULL, 10);
+
+	if (((argc < 2) && (firmware_target != DFU_GECKO)) || 
+			((argc != 3) && (firmware_target == DFU_GECKO))) {
 		shell_error(shell, "Missing required arguments");
-		shell_print(shell, "Usage: tmo dfu update <target> <modem delta file>\n"
-				"       target          : 0 for mcu, 1 for modem, 2 for wifi/ble\n"
-				"       modem delta file: SAMPLE: 0 for update3.2.20351_20161_dis.ua, 1 for update3.2.20161_20351_dis.ua\n"
-				"                         GOLDEN: 2 for update20351_20161_801_test.ua, 3 for update20161_20351_801_test.ua\n");
+		shell_print(shell, "Usage: tmo dfu update <target>\n"
+				"       target : 0 for mcu, 1 for modem, 2 for wifi/ble\n"
+				"       mcu_slot(optional): slot to update. Applicable only to mcu target\n"
+				"       Usage (mcu): \n"
+				"                   tmo dfu update 0 [slot]\n"
+				"       Usage (modem): \n"
+				"                   tmo dfu update 1 [modem_delta_file]\n"
+				"       Usage (wifi/ble): \n"
+				"                   tmo dfu update 2\n");
 		return -EINVAL;
 	}
 
-	int firmware_target = (int) strtol(argv[1], NULL, 10);
-	int delta_firmware_target = (int) strtol(argv[2], NULL, 10);
-
 	switch (firmware_target)
 	{
 		case DFU_GECKO:
 			{
+#ifdef BOOT_SLOT
+				if ((strcmp(BOOT_SLOT, "0") || strcmp(BOOT_SLOT, "1"))) {
+					if (atoi(BOOT_SLOT) == delta_firmware_target) {
+						shell_error(shell,"Can't program slot you are running from");
+						return -EINVAL;
+					}
+				}
+#else
+				if (delta_firmware_target != 1) {
```

Comment:
i think this line should be removed so that it always returns an error when trying to update a slot without bootloader

Reply:
That is the way it use to work but Kevin had me add that line so that we could program if there is no slot being used.